src/collectors/dmi_collector.rs

```diff
@@ -0,0 +1,282 @@
+    let mut split: Vec<&str> = Vec::new();
+
+    let mut system_information: SystemInformation = SystemInformation {
+        vendor: String::new(),
+        model: String::new(),
+        uuid: String::new(),
+        serial: String::new(),
+    };
+
+    let mut table_found: bool = false;
+
+    for part in output_split {
+        if !table_found {
+            table_found = find_table("System Information", part);
+        }
+
+        let split_output: Result<Vec<&str>, &str> = split_output(part);
+
+        match split_output {
+            Ok(_) => split = split_output.unwrap(),
+            Err(_) => continue,
+        }
+
+        let mut key: String = String::new();
+        let mut value: String = String::new();
+
+        match split.get(0) {
+            Some(x) => {
+                key = x.to_string();
+            }
+            None => println!("Key not found!")
+        }
+        match split.get(1) {
+            Some(x) => {
+                value = x.to_string();
+            }
+            None => println!("Value not found!")
+        }
+        match key.as_str() {
+            "Manufacturer" => {
+                system_information.vendor = value.trim().to_string();
+            },
+            "Product Name" => {
+                system_information.model = value.trim().to_string();
+            },
+            "Serial Number" => {
+                system_information.serial = value.trim().to_string();
+            },
+            "UUID" => {
+                system_information.uuid = value.trim().to_string();
+            },
+            _=> {
+                continue;
+            },
+        }
+    }
+    return system_information
+}
+
+
+fn dmidecode_chassis() -> ChassisInformation {
+    /*
+     * Collect Chassis information.
+     *
+     * Works like dmidecode_system with only one key to be checked.
+     * */
+    let output: String = execute_dmidecode(3);
+    let output_split: Split<'_, &str> = output.split("\n");
+    let mut split: Vec<&str> = Vec::new();
+
+    let mut chassis_information: ChassisInformation = ChassisInformation {asset: String::new()};
+
+    let mut table_found: bool = false;
+
+    for part in output_split {
+        if !table_found {
+            table_found = find_table("System Information", part);
+        }
+
+        let split_output: Result<Vec<&str>, &str> = split_output(part);
+
+        match split_output {
+            Ok(_) => split = split_output.unwrap(),
+            Err(_) => continue,
+        }
+
+        let mut key: String = String::new();
+        let mut value: String = String::new();
+
+        match split.get(0) {
+            Some(x) => {
+                key = x.to_string();
+            }
+            None => println!("Info: Key not found at this location...")
+        }
+        match split.get(1) {
+            Some(x) => {
+                value = x.to_string();
+            }
+            None => println!("Info: Key not found at this location...")
```

Comment:
The error message should be about the value instead of the key

Reply:
Thanks for the heads up!